File: bms/go/go-gt/delta_part/delta_part.go
package test

var ()

const ()

func delta_part(a, b, dist *Matrix, p Vector, i, j, r, s int64) int64 {
	return (dist.Get(i, j) + (a.Get(r, i)-a.Get(r, j)+a.Get(s, j)-a.Get(s, i))*
		(b.Get(p[s], p[i])-b.Get(p[s], p[j])+b.Get(p[r], p[j])-b.Get(p[r], p[i])) +
		(a.Get(i, r)-a.Get(j, r)+a.Get(j, s)-a.Get(i, s))*
			(b.Get(p[i], p[s])-b.Get(p[j], p[s])+b.Get(p[j], p[r])-b.Get(p[i], p[r])))
}

func (m Matrix) Get(i int64, j int64) int64 {
	return m.A[i*m.N+j]
}

type Vector []int64

type Matrix struct {
	N int64
	A []int64
}

File: translations_only/success/go/go-gt/delta_part/claude2/BaseRepair/delta_part.rs


use std::ops::IndexMut;

struct Vector(Vec<usize>);

struct Matrix {
    n: usize,
    a: Vec<i64>,
}

fn get(matrix: &Matrix, i: usize, j: usize) -> i64 {
    matrix.a[i * matrix.n + j]  
}

fn delta_part(
    a: &Matrix,
    b: &Matrix,
    dist: &Matrix,
    p: &Vector,
    i: usize,
    j: usize, 
    r: usize,
    s: usize) -> i64 {

    let p_s = p.0[s];
    let p_i = p.0[i];
    let p_j = p.0[j];
    let p_r = p.0[r];
    
    dist.a[i * dist.n + j] +
        (get(a, r, i) - get(a, r, j) + get(a, s, j) - get(a, s, i)) *
        (get(b, p_s, p_i) - get(b, p_s, p_j) + get(b, p_r, p_j) - get(b, p_r, p_i)) + 
        (get(a, i, r) - get(a, j, r) + get(a, j, s) - get(a, i, s)) * 
        (get(b, p_i, p_s) - get(b, p_j, p_s) + get(b, p_j, p_r) - get(b, p_i, p_r))
}

